lib/Handlers/ifV2.ts

```diff
@@ -0,0 +1,48 @@
+import { NodeType } from '@voiceflow/general-types/';
+import { Node } from '@voiceflow/general-types/build/nodes/ifV2';
+
+import { HandlerFactory } from '@/lib/Handler';
+
+import { TurnType } from '../Constants/flags';
+import CodeHandler from './code';
+
+export type IfV2Options = {
+  endpoint?: string | null;
+  safe?: boolean;
+};
+
+const IfV2Handler: HandlerFactory<Node, IfV2Options | void> = ({ endpoint, safe } = {}) => ({
+  canHandle: (node, runtime) => node.type === NodeType.IF_V2 && !(runtime.turn.get<string[]>(TurnType.STOP_TYPES) || []).includes(NodeType.IF_V2),
+  handle: async (node, runtime, variables, program) => {
+    let outputPortIndex = -1;
+    const setOutputPort = function(index: number) {
+      outputPortIndex = index;
+    };
+    const codeHandler = CodeHandler({ endpoint, callbacks: { setOutputPort }, safe });
+
+    let code = '';
+    for (let i = 0; i < node.payload.expressions.length; i++) {
+      code += `
+            if(${node.payload.expressions[i]}) {
+                setOutputPort(${i}); 
+                throw(null);
+            }
+        `;
+    }
+
+    const codeTemplate = `try { ${code} } catch (err) {}`;
```

Comment:
maybe we can add a debug trace for errors?

Reply:
this try catch is mostly needed to exit early once a condition is met, the code block already adds a trace for failures of the whole execution